test-utils/runtime-tester/src/fuzzing.rs

```diff
@@ -0,0 +1,174 @@
+use crate::run_test::{BlockConfig, NetworkConfig, Scenario, TransactionConfig};
+use near_crypto::{InMemorySigner, KeyType};
+use near_primitives::{
+    account::{AccessKey, AccessKeyPermission},
+    transaction::{Action, AddKeyAction, CreateAccountAction, StakeAction, TransferAction},
+    types::{AccountId, Balance, BlockHeight, Nonce},
+};
+use nearcore::config::{NEAR_BASE, TESTING_INIT_BALANCE};
+
+use libfuzzer_sys::arbitrary::{Arbitrary, Result, Unstructured};
+
+impl Arbitrary<'_> for Scenario {
+    fn arbitrary(u: &mut Unstructured<'_>) -> Result<Self> {
+        let num_accounts = u.int_in_range(2..=100)?;
+
+        let seeds: Vec<String> = (0..num_accounts).map(|i| format!("test{}", i)).collect();
+
+        let mut scope = Scope::from_seeds(&seeds);
+
+        let network_config = NetworkConfig { seeds };
+
+        let num_blocks = u.int_in_range(1500..=2500)?;
+
+        let mut blocks = vec![];
+
+        for _ in 0..num_blocks {
+            blocks.push(BlockConfig::arbitrary(u, &mut scope)?);
+        }
+        Ok(Scenario { network_config, blocks })
+    }
+}
+
+impl BlockConfig {
+    fn arbitrary(u: &mut Unstructured, scope: &mut Scope) -> Result<BlockConfig> {
+        scope.inc_height();
+        let mut block_config = BlockConfig::at_height(scope.get_height());
+
+        let num_txs = u.int_in_range(0..=500)?;
```

Comment:
Same here, need to choose number of txs based on u.len() (and probably some hardcoded max number of txs in a block)